projects/novo-elements/src/elements/data-table/state/data-table-state.service.ts

```diff
@@ -23,8 +23,9 @@ export class DataTableState<T> {
   expandedRows: Set<string> = new Set<string>();
   outsideFilter: any;
   isForceRefresh: boolean = false;
-
+  selectionOptions: IDataTableSelectionOption[];
   updates: EventEmitter<IDataTableChangeEvent> = new EventEmitter<IDataTableChangeEvent>();
+  retainSelected: boolean = false;
 
```

Comment:
this might be something we want to define at the data-table level as opposed to a service level feature

Reply:
This doesn't really drive the retention, it is used in the state service to determine if the next event will retain the selected values. It's pretty much only used in the state service via the checkRetainment method. Should we still move this out to the component?